providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -159,33 +159,41 @@ private boolean isConnected() {
     }
 
     private void onConnectionEvent(ConnectionEvent connectionEvent) {
-        boolean previous = connected;
-        boolean current = connected = connectionEvent.isConnected();
+        final boolean wasConnected = connected;
```

Comment:
[suggestion] We are storing the application's state in a' connected' boolean and mapping our ConnectionState to it. Would it be worth changing the type of connection and working with the state directly? This whole method looks like a state machine, and I think handling this with a state-like pattern could be beneficial, like 

```java
private ConnectionState connectionState;

private void onConnectionEvent(ConnectionEvent connectionEvent) {
    if (!isInitialized) {
          return;
    }
    switch(connectionEvent.getState()) {
        case ConnectionState.CONNECTED:
                 var builder = ProviderEventDetails.builder()
                    .flagsChanged(connectionEvent.getFlagsChanged())

                 if(connectionState != ConnectionState.CONNECTED) {
                     this.emitProviderReady(builder.message("connected to flagd").build());
                 } else if (connectionState == ConnectionState.CONNECTED) {
                     this.emitProviderConfigurationChanged("configuration changed").build());
                 }
                // ....
            break;
        case ConnectionState.STALE:
                   this.emitProviderStale(ProviderEventDetails.builder().message("there has been an error").build());
            break;
        case ConnectionState.ERROR:
                  this.emitProviderError(ProviderEventDetails.builder().message("there has been an error").build());
            break;
    }
    this.connectionState = connectionEvent.getState();
}
```
just an idea/suggestion - no need to act

Reply:
Agreed this seems "nice" but not required for me.